api/users/views.py

```diff
@@ -983,3 +994,80 @@ class UserMessageView(JSONAPIBaseView, generics.CreateAPIView):
 
     view_category = 'users'
     view_name = 'user-messages'
+
+
+class ExternalLoginConfirmEmailView(generics.CreateAPIView):
```

Comment:
Include a csrf token check as well, just to ensure the user in on the login page. A user can use CAS to do this via the Back-end alone, but this route is for Front-end only.